From 2b6b64e545a8ef46e4c43113e80898400c6f3fc0 Mon Sep 17 00:00:00 2001
From: Russ Weight <russell.h.weight@intel.com>
Date: Mon, 5 Apr 2021 11:13:31 -0700
Subject: [PATCH] configs: DFL driver config options

Having separate config files means having a lot of duplicate options
listed in both files. It also means that both config files need to be
used in order to build all of the DFL support into a single kernel.

This change merges the configs into a single list of config options and
renames the readme.txt file to README.

The n3000_d5005_defconfig and n5010_defconfig files are now deprecated
and will be removed soon.

Signed-off-by: Russ Weight <russell.h.weight@intel.com>
---
 configs/README     | 13 +++++++++++
 configs/dfl-config | 56 ++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 69 insertions(+)
 create mode 100644 configs/README
 create mode 100644 configs/dfl-config

diff --git a/configs/README b/configs/README
new file mode 100644
index 000000000000..d100a073056b
--- /dev/null
+++ b/configs/README
@@ -0,0 +1,13 @@
+This directory contains the dfl-config file which includes the config
+options required to enable the Device Feature List (DFL) driver support.
+By design the DFL driver collection is extendable and it is anticipated
+that new drivers will be added to the collection.
+
+The dfl related config options are intended to be appended to a base kernel
+configuration. For example the following commands would modify an existing
+kernel configuration to enable support DFL cards.
+
+	# cd kernel_source_directory
+	# cp /boot/config-`uname -r` .config
+	# cat configs/dfl-config >> .config
+	# make olddefconfig
diff --git a/configs/dfl-config b/configs/dfl-config
new file mode 100644
index 000000000000..944d3746279c
--- /dev/null
+++ b/configs/dfl-config
@@ -0,0 +1,56 @@
+# Core FPGA support
+
+CONFIG_FPGA=m
+CONFIG_FPGA_BRIDGE=m
+CONFIG_FPGA_REGION=m
+
+# Core DFL support
+
+CONFIG_FPGA_DFL=m
+CONFIG_FPGA_DFL_PCI=m
+CONFIG_FPGA_DFL_AFU=m
+CONFIG_FPGA_DFL_FME=m
+CONFIG_FPGA_DFL_FME_MGR=m
+CONFIG_FPGA_DFL_FME_BRIDGE=m
+CONFIG_FPGA_DFL_FME_REGION=m
+
+CONFIG_SPI_ALTERA=m
+CONFIG_SPI_ALTERA_DFL=m
+CONFIG_SERIAL_8250_DFL=m
+CONFIG_FPGA_M10_BMC_SEC_UPDATE=m
+
+CONFIG_MFD_INTEL_M10_BMC_SPI=m
+CONFIG_SENSORS_INTEL_M10_BMC_HWMON=m
+
+CONFIG_FW_UPLOAD=y
+
+CONFIG_UIO_DFL=m
+
+# N3000 / D5005 Support
+
+CONFIG_MEMORY=y
+CONFIG_IIO_INTERRUPT_TRIGGER=m
+
+CONFIG_FPGA_DFL_EMIF=m
+CONFIG_FPGA_DFL_NIOS_INTEL_PAC_N3000=m
+CONFIG_FPGA_DFL_HSSI=m
+CONFIG_S10HSSI=m
+CONFIG_INTEL_S10_PHY=m
+
+# N5010 Support
+
+CONFIG_FIXED_PHY=y
+CONFIG_N5010_HSSI=m
+CONFIG_N5010_PHY=m
+CONFIG_NET_VENDOR_SILICOM=y
+
+# PMCI device support
+
+CONFIG_MFD_INTEL_M10_BMC_LOG=m
+CONFIG_MFD_INTEL_M10_BMC_PMCI=m
+CONFIG_QSFP_MEM=m
+CONFIG_QSFP_MEM_DFL=m
+
+# ToD device support
+
+CONFIG_PTP_DFL_TOD=m
